components/brave_wallet/browser/internal/hd_key_sr25519.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_SR25519_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_SR25519_H_
+
+#include <memory>
+
+#include "base/containers/span.h"
+#include "brave/components/brave_wallet/browser/internal/polkadot/rust/sr25519.h"
+
+namespace brave_wallet {
+
+namespace polkadot {
+class SchnorrkelKeyPair;
+}  // namespace polkadot
+
+class HDKeySr25519 {
```

Comment:
I'm not sure if comments are necessary, but please review the style guide and use of good judgement to determine where to add comments in this class header:

https://google.github.io/styleguide/cppguide.html#Class_Comments
https://google.github.io/styleguide/cppguide.html#Function_Comments
https://google.github.io/styleguide/cppguide.html#Variable_Comments

Reply:
Good looking out. This is something clang-format had maybe added once but never removed as the code was refactored. I'm not sure I remember adding it myself. But you are right in that our current clang-format file doesn't automatically add the closing comment until there's two lines in the enclosing namespace.

Other places in the wallet code seem to use the closing comment for nested namespaces.

Reply:
This is done!

Reply:
I'm reopening this thread because I think there's still some margin in improving the comments here. Take a look for instance on how documented `crypto::keypair::PrivateKey` is documented in chromium

https://source.chromium.org/chromium/chromium/src/+/main:crypto/keypair.h

Reply:
lmao I thought you were talking about the trailing comment closing the namespace

Now I realize what you mean: actually comment the code out like the other HDKey classes  in the same directory are.